app/Http/Controllers/PostsController.php

```diff
@@ -96,9 +117,33 @@ public function edit($id)
      */
     public function update(Request $request, $id)
     {
+
+        $this->validate($request,[
+            'title'=>'required',
+            'body'=>'required',
+            'cover_image'=>'image|nullable|max:1999'
+        ]);
+
+        //Handle File uploading 
+        if ($request->hasfile('cover_image')){
+            //Get file name with the extension
+            $fileNameWithExt = $request->file('cover_image')->getClientOriginalName();
+            //Get just file name
+            $fileName = pathinfo($fileNameWithExt, PATHINFO_FILENAME);
+            //Get just Ext
+            $extension = $request->file('cover_image')->getClientOriginalExtension();
+            //File name to store
+            $fileNameToStore = $fileName.'_'.time().'.'.$extension;
+            //Upload image
+            $path = $request->file('cover_image')->storeAs('public/cover_images', $fileNameToStore);
+        }
+
         $post = Post::find($id);
         $post->title = $request->input('title');
         $post->body = $request->input('body');
+        if ($request->hasFile('cover_image')){
```

Comment:
This part is needs to be rethink. Because it doesn't let you pick NoImage.jpg instead of your original image.